viewing_party/party.py

```diff
@@ -2,22 +2,178 @@
+    else:
+        for i in range(len(user_data["watched"])):
+            ratings.append((user_data["watched"][i]["rating"]))
+            i+=1
+    
+        for rating in ratings:
+            ratings_sum += rating
+        
+        ratings_average = ratings_sum / len(ratings)
+
+    return ratings_average  
+
+def get_most_watched_genre(user_data):
+    genre_list = []
+    genre_dictionary = {}
+
+    if (len(user_data["watched"])) is 0:
+        popular_genre = None
+    else:
+        for i in range(len(user_data["watched"])):
+            genre_list.append((user_data["watched"][i]["genre"]))
+            i += 1
+    
+    for genre in genre_list:
+        count = genre_list.count(genre)
+        genre_dictionary[genre] = count
+
+    try:    
+        popular_genre = max(genre_dictionary, key=genre_dictionary.get)
+    except ValueError as err:
+        popular_genre = None
+
+
+    return(popular_genre)
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    movie_titles_user_watched = set()
+    movie_titles_friends_watched = set()
+    list_of_movie_dictionaries_friends_have_not_watched = []
+
+    for movie in user_data["watched"]:
+        movie_titles_user_watched.add((movie['title']))
+    
+    for i in range(len(user_data["friends"])):
+        for movie in user_data["friends"][i]["watched"]:
+            title = movie.get("title")
+            movie_titles_friends_watched.add(title)
+        i += 1 
+    
+    movies_user_has_watched_but_not_friends = movie_titles_user_watched.difference(movie_titles_friends_watched)
+
+    for movie in movies_user_has_watched_but_not_friends:
+        for dictionary in user_data["watched"]:
+            if movie == dictionary['title']:
+                list_of_movie_dictionaries_friends_have_not_watched.append(dictionary)
+
+    return(list_of_movie_dictionaries_friends_have_not_watched)
+
+def get_friends_unique_watched(user_data):
+    movie_titles_friends_watched = set()
+    movie_titles_user_watched = set()
+    list_of_movie_dictionaries_user_has_not_watched = []
+
+    for i in range(len(user_data["friends"])):
+        for movie in user_data["friends"][i]["watched"]:
+            title = movie.get("title")
+            movie_titles_friends_watched.add(title)
+        i += 1
+    
+    for movie in user_data["watched"]:
+        movie_titles_user_watched.add((movie["title"]))
+    
+    movies_friends_have_watched_but_not_user = movie_titles_friends_watched.difference(movie_titles_user_watched)
+
+    for movie in movies_friends_have_watched_but_not_user:
+        for i in range(len(user_data["friends"])):
+            for dictionary in user_data["friends"][i]["watched"]:
+                title = dictionary.get("title")
+                if title == movie:
+                    list_of_movie_dictionaries_user_has_not_watched.append(dictionary)
+            i+=1

```

Comment:
hmm where have we seen this code already? maybe we could turn this into a helper function that we call upon inside this function.